src/OpenSage.Game/Network/SkirmishSlot.cs

```diff
@@ -26,18 +35,52 @@ public SkirmishSlot(int index)
 
         public int Index { get; set; }
         public SkirmishSlotState State { get; set; } = SkirmishSlotState.Open;
-        public string PlayerName { get; set; }
+        public string PlayerName { get; set; } = string.Empty;
         public byte ColorIndex { get; set; }
         public byte FactionIndex { get; set; }
         public byte Team { get; set; }
         public bool Ready { get; set; }
-        public IPEndPoint EndPoint { get; set; }
+
+        public IPEndPoint EndPoint;
 
         /// <summary>
         /// We need this during development to be able to run two games on the same machine.
         /// </summary>
         public int ProcessId { get; set; }
 
+        private byte[] _cleanState = new byte[1];
+        public bool IsDirty {
+            get
+            {
+                return !_cleanState.SequenceEqual(_bytes);
+            }
+            internal set {
+                if (!value)
+                {
+                    _cleanState = _bytes;
+                }
+            }
+        }
+
+        private byte[] _bytes
+        {
+            get
+            {
+                using (var ms = new MemoryStream())
+                {
+                    var writer = new BinaryWriter(ms);
+                    writer.Write(Index);
+                    writer.Write((int)State);
+                    writer.Write(PlayerName);
+                    writer.Write(ColorIndex);
+                    writer.Write(FactionIndex);
+                    writer.Write(Team);
+                    writer.Write(Ready);
+                    return ms.ToArray();
+                }
+            }
+        }
```

Comment:
That was the other option. I chose against it, because this was less lines of code, and less repeating code (thus more beautiful) in my opinion. This is slower, but it's only done every 100ms anyway.